File: src/main/java/org/tud/oas/api/accessibility/AccessibilityController.java
package org.tud.oas.api.accessibility;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import org.tud.oas.accessibility.Accessibility;
import org.tud.oas.accessibility.GravityAccessibility;
import org.tud.oas.accessibility.MultiCriteraAccessibility;
import org.tud.oas.accessibility.PopulationAccessibility;
import org.tud.oas.accessibility.SimpleAccessibility;
import org.tud.oas.population.Population;
import org.tud.oas.population.PopulationManager;
import org.tud.oas.population.PopulationView;
import org.tud.oas.routing.IRoutingProvider;
import org.tud.oas.routing.RoutingManager;

@RestController
@RequestMapping("/v1/accessibility")
public class AccessibilityController {

    @PostMapping("")
    public GridResponse calcGrid(@RequestBody SimpleAccessibilityRequest request) throws Exception {
        return this.calcSimpleGrid(request);
    }

    @PostMapping("/simple")
    public GridResponse calcSimpleGrid(@RequestBody SimpleAccessibilityRequest request) throws Exception {
        Population population = PopulationManager.getPopulation();
        IRoutingProvider provider = RoutingManager.getRoutingProvider();

        SimpleAccessibility simple = new SimpleAccessibility(population, provider);

		simple.calcAccessibility(request.getLocations(), request.getRanges());
        
        return simple.buildResponse();
    }

    @PostMapping("/gravity")
    public GridResponse calcGravityGrid(@RequestBody GravityAccessibilityRequest request) throws Exception {
        Population population = PopulationManager.getPopulation();
        IRoutingProvider provider = RoutingManager.getRoutingProvider();
        PopulationView view = population.getPopulationView(request.getEnvelop());

        long start = System.currentTimeMillis();
        GravityAccessibility gravity = new GravityAccessibility(view, provider);

		gravity.calcAccessibility(request.getLocations(), request.getRanges(), request.getFactors());
        long end = System.currentTimeMillis();

        System.out.println("time: " + (end - start));

        return gravity.buildResponse();
    }

    @PostMapping("/multi")
    public GridResponse calcMultiCriteriaGrid(@RequestBody MultiCriteriaRequest request) throws Exception {
        Population population = PopulationManager.getPopulation();
        IRoutingProvider provider = RoutingManager.getRoutingProvider();
        PopulationView view = population.getPopulationView(request.getEnvelope());

        view.setPopulationType(request.getPopulationType());
        view.setPopulationIndizes(request.getPopulationIndizes());

        GravityAccessibility gravity = new GravityAccessibility(view, provider);

        MultiCriteraAccessibility multiCriteria = new MultiCriteraAccessibility(view, gravity);

        for (Map.Entry<String, InfrastructureParams> entry : request.getInfrastructures().entrySet()) {
            InfrastructureParams value = entry.getValue();
            multiCriteria.addAccessibility(entry.getKey(), value.getLocations(), value.getRanges(), value.getFactors(), value.getWeight());
        }
        multiCriteria.calcAccessibility();
        return multiCriteria.buildResponse();
    }
}

File: src/api/accessibility/AccessibilityController.cs
using System;
using System.Runtime.InteropServices;
using Microsoft.AspNetCore.Http;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using DVAN.Routing;
using System.Threading.Tasks;
using DVAN.Accessibility;
using DVAN.Population;

namespace DVAN.API
{
    [ApiController]
    [Route("/v1/accessibility")]
    public class AccessibilityController {

        [HttpPost]
        public async Task<GridResponse> calcGrid([FromBody] SimpleAccessibilityRequest request)
        {
            return await this.calcSimpleGrid(request);
        }

        [HttpPost("simple")]
        public async Task<GridResponse> calcSimpleGrid([FromBody] SimpleAccessibilityRequest request)
        {
            PopulationContainer population = PopulationManager.getPopulation();
            IRoutingProvider provider = RoutingManager.getRoutingProvider();

            SimpleAccessibility simple = new SimpleAccessibility(population, provider);

            await simple.calcAccessibility(request.getLocations(), request.getRanges());
            
            return simple.buildResponse();
        }

        [HttpPost("gravity")]
        public async Task<GridResponse> calcGravityGrid([FromBody] GravityAccessibilityRequest request)
        {
            PopulationContainer population = PopulationManager.getPopulation();
            IRoutingProvider provider = RoutingManager.getRoutingProvider();
            PopulationView view = population.getPopulationView(request.getEnvelop());

            long start = Environment.TickCount64;
            GravityAccessibility gravity = new GravityAccessibility(view, provider);

            await gravity.calcAccessibility(request.getLocations(), request.getRanges(), request.getFactors());
            long end = Environment.TickCount64;

            Console.WriteLine("time: " + (end - start));

            return gravity.buildResponse();
        }

        [HttpPost("multi")]
        public async Task<GridResponse> calcMultiCriteriaGrid([FromBody] MultiCriteriaRequest request) 
        {
            PopulationContainer population = PopulationManager.getPopulation();
            IRoutingProvider provider = RoutingManager.getRoutingProvider();
            PopulationView view = population.getPopulationView(request.getEnvelope());

            view.setPopulationType(request.getPopulationType());
            view.setPopulationIndizes(request.getPopulationIndizes());

            GravityAccessibility gravity = new GravityAccessibility(view, provider);

            MultiCriteraAccessibility multiCriteria = new MultiCriteraAccessibility(view, gravity);

            foreach (var entry in request.getInfrastructures()) {
                InfrastructureParams value = entry.Value;
                await multiCriteria.addAccessibility(entry.Key, value.facility_locations, value.ranges, value.range_factors, value.infrastructure_weight);
            }
            multiCriteria.calcAccessibility();
            var response = multiCriteria.buildResponse();
            return response;
        }
    }
}